src/utils.js

```diff
@@ -236,6 +236,91 @@ function camelcase (dashCaseText) {
     })
 }
 
+function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+
+    // Best guess if the device is an Apple Silicon
+    // https://stackoverflow.com/a/65412357
+    return gl.getSupportedExtensions().indexOf('WEBGL_compressed_texture_etc') !== -1
+}
+
+/**
+ * Take configSeting which should be an array of possible values.
+ * If a value contains a criteria that is a match for this environment then return that value.
+ * Otherwise return the first value that doesn't have a criteria.
+ *
+ * @param {array} configSetting - Config setting which should contain a list of possible values
+ * @returns - The value from the list that best matches the criteria in the config
+ */
+function processAttrByCriteria (configSetting) {
+    if (!configSetting.some(item => item.criteria !== undefined)) {
+        return configSetting
+    }
+
+    let bestOption = null
+    for (const item of configSetting) {
+        if (item.criteria) {
+            if (item.criteria.arch === 'AppleSilicon' && isAppleSilicon()) {
+                bestOption = item
+                break
+            }
+        } else {
+            bestOption = item
+        }
+    }
+
+    if (bestOption && bestOption.type) {
```

Comment:
I think this can be restructured to return just the best option and _getFeatureAttr_ handle the undefined and value code as it already does.

Reply:
Yeah not very DRY. Fixed.